### DISCUSSION TITLE
Connect MPG directly to Nucleo for X,Y, and Z axis control

### DISCUSSION THREAD

**Discussion by yuuahmad** [grblHAL/core#437] [Q&A]
@terjeio Is there an easy way (for example using a plugin) so that MPG can be read by Nucleo and used to move the x, y or z axis?

The schematic that I proposed can be seen in the image that I included.

I have read and understood the [encoder plugin](https://github.com/grblHAL/Plugin_encoder.git) but can't find the features I mean and want.

because the MPG in this feature can only be used to regulate (override) feed rate, rapid rate and spindle rpm.

![MPG to Nucleo](https://github.com/grblHAL/core/assets/22028802/27e8b29b-9001-498b-ba89-7d2277be4612)

**Comment by terjeio** [comment]
There are two jog modes in the code but these are crude and may not work at all. Currently these cannot be enabled without changing the code.
I you want to try you have to [add option(s) to $400](https://github.com/grblHAL/Plugin_encoder/blob/9a387e21f40bb9a35b8f6d9c1313f5997beccbf0/encoder.c#L619) by extending the comma separated list - new values should match [this enum](https://github.com/grblHAL/core/blob/4a7090d70aadddcd426d5326bb252fe4dc3ee40c/plugins.h#L158-L171).
Switching jog mode can be done [here](https://github.com/grblHAL/Plugin_encoder/blob/9a387e21f40bb9a35b8f6d9c1313f5997beccbf0/encoder.c#L756-L757).

FYI more advanced solutions such as [this](https://github.com/dresco/grblPANEL) and [this](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack) uses a second microcontroller to provide encoder based jogging etc.

**Reply by terjeio** [reply]
Comment deleted.

**Reply by terjeio** [reply]
Sorry, I posted the previous comment in the wrong thread.

**Comment by nickshl** [comment]
> FYI more advanced solutions such as [this](https://github.com/dresco/grblPANEL)

Huge!

>and [this](https://github.com/terjeio/GRBL_MPG_DRO_BoosterPack) uses a second microcontroller to provide encoder based jogging etc.

Huge!

If it doesn't fit into your hand - it is this:
![New_Zealand_Rotary_Telephone](https://github.com/grblHAL/core/assets/38679154/00ca7713-0e61-412e-beb7-8d953f578191)
Not this:
![IPhone_First_Generation_8GB_(3677961514)](https://github.com/grblHAL/core/assets/38679154/86e996a5-71b6-419e-81e7-614d0c4d6bfe)
Soon... hopefully ;)
